app.js

```diff
@@ -0,0 +1,485 @@
+  el.coinsContainer.addEventListener("click", (ev) => {
+    const btn = ev.target.closest(".toggle-btn"); if (!btn) return;
+    const sym = btn.getAttribute("data-symbol"); if (!sym) return;
+
+    const idx = selected.indexOf(sym);
+    if (idx >= 0) {
+      selected.splice(idx, 1);
+      btn.classList.remove("active");
+      btn.textContent = "OFF";
+      persistSelected();
+      if (!el.reportsSection.classList.contains("d-none")) startReports();
+    } else {
+      if (selected.length < 5) {
+        selected.push(sym);
+        btn.classList.add("active");
+        btn.textContent = "ON";
+        persistSelected();
+        if (!el.reportsSection.classList.contains("d-none")) startReports();
+      } else {
+        openReplaceModal(sym);
+      }
+    }
+  });
+
+  function openReplaceModal(requested) {
+    el.limitList.replaceChildren();
+    selected.forEach((s, i) => {
+      const label = document.createElement("label");
+      label.className = "list-group-item d-flex justify-content-between align-items-center";
+      label.innerHTML = `
+        <div class="d-flex align-items-center gap-2">
+          <input class="form-check-input me-1" type="radio" name="to-remove" value="${s}" ${i === 0 ? "checked" : ""}>
+          <span>${s}</span>
+        </div>
+        <span class="badge bg-primary rounded-pill">in report</span>`;
+      el.limitList.appendChild(label);
+    });
+    el.limitConfirm.onclick = () => {
+      const toRemove = el.limitList.querySelector("input[name='to-remove']:checked")?.value;
+      if (!toRemove) return;
+      selected = selected.filter((x) => x !== toRemove);
+      selected.push(requested);
+      persistSelected();
+      document.querySelectorAll(".toggle-btn").forEach((b) => {
+        const s = b.getAttribute("data-symbol");
+        const on = selected.includes(s);
+        b.classList.toggle("active", on);
+        b.textContent = on ? "ON" : "OFF";
+      });
+      bootstrap.Modal.getOrCreateInstance(el.limitModal).hide();
+      if (!el.reportsSection.classList.contains("d-none")) startReports();
+    };
+    bootstrap.Modal.getOrCreateInstance(el.limitModal).show();
+  }
+
+  document.addEventListener("show.bs.collapse", async (ev) => {
+    const wrap = ev.target.closest("[data-coin-id]");
+    if (!wrap) return;
+    const coinId = wrap.getAttribute("data-coin-id");
+    const container = wrap.querySelector(".moreinfo-body");
+    if (!coinId || !container) return;
+
+    const cached = infoCache[coinId];
+    if (cached && Date.now() - cached.ts < TTL_INFO_MS) {
+      renderMoreInfo(container, cached.data, true);
+      return;
+    }
+    try {
+      const data = await loadAssetById(coinId);
+      infoCache[coinId] = { data, ts: Date.now() };
+      renderMoreInfo(container, data, false);
+    } catch {
+      container.innerHTML = `<div class="alert alert-danger m-0">Error loading details.</div>`;
+    }
+  });
+
+  async function renderMoreInfo(container, data, fromCache) {
```

Comment:
this should have been done in a floater/tooltip to prevent flickering

<img width="1320" height="650" alt="Screenshot 2025-09-10 at 12 34 46" src="https://github.com/user-attachments/assets/7fac33d4-84b7-4702-928a-9ba05ce58633" />